client/product_details/AddToBagButton.jsx

```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+const AddToBagButton = (props) => {
+  // if there is no stock, the button will not render
+  if (props.availableSizes.length !== 0) {
+    return (
+      <button id="addToBagButton" onClick={props.handleAddToBagSubmit}>ADD TO BAG</button>
+    );
+  }
+  return null;
```

Comment:
I don't know why but I've never returned null and added some logic in functional components this way before. Seems like a fantastic way to not clog up top components with complex and unnecessary code!